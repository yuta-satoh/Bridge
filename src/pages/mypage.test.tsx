import renderer from 'react-test-renderer';
import Mypage from './mypage';



// テスト用のクッキーの値を定義
const testCookieValue = '1';
describe('mypage.tsxの描画テスト', () => {
  it('ロード中', () => {
    // cookieValueにテスト用のクッキーの値を格納
    const cookieValue = testCookieValue;

    // propsにcookieValueを定義
    const props = { cookieValue };

    // Mypageコンポーネントにpropsを渡す
    const mypage_component = <Mypage {...props} />;

    // 描画テスト
    const tree = renderer.create(mypage_component).toJSON();
    expect(tree).toMatchSnapshot();
  });


});
