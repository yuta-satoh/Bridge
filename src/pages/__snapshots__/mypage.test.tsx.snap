// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mypage.tsxのスナップショットテスト ロード中 1`] = `
<main
  className="main"
>
  <section
    className="section"
  >
    <div
      className="head"
    >
      <nav>
        <ol
          className="links"
          id="top"
        >
          <li
            className="pageLink"
          >
            <a
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              Bridge
            </a>
            <span
              className="greaterThan"
            >
              &gt;
            </span>
          </li>
          <li
            className="pageLink"
          >
            マイページ
          </li>
        </ol>
      </nav>
      <h1
        className="title"
      >
        マイページ
      </h1>
    </div>
    <div
      className="body"
    >
      <div
        className="container"
      >
        <div
          className="cartTitle"
        >
          <div>
            <h2>
              カートの商品
            </h2>
          </div>
        </div>
        <div
          className="cartList"
        >
          <div
            className="container"
          >
            <div
              className="background"
            >
              <p>
                ロード中...
              </p>
            </div>
          </div>
        </div>
        <div
          className="cartLink"
        >
          <a
            href="/cart"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            &gt;&gt;カート一覧
          </a>
        </div>
      </div>
      <div
        className="container"
      >
        <div
          className="subtitle"
        >
          <h2>
            会員メニュー
          </h2>
        </div>
        <div
          className="updateProfArea"
        >
          <button
            className="updateProfButton"
            onClick={[Function]}
            type="button"
          >
            会員情報の確認/変更
          </button>
          <button
            className="updateProfButton"
            onClick={[Function]}
            type="button"
          >
            パスワードの変更
          </button>
          <button
            className="updateProfButton"
            onClick={[Function]}
            type="button"
          >
            購入履歴
          </button>
        </div>
        <div
          className="logout"
        >
          &gt;&gt;
          <div
            className="logout"
          >
            <a
              href="/login"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              ログアウト
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</main>
`;
